src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs

```diff
@@ -12,11 +13,56 @@ namespace Microsoft.IdentityModel.Tokens
     /// </summary>
     public class SecurityTokenDescriptor
     {
+        // TODO: At next major version (8.0), remove Audience and logic for combining with Audiences.
+        private HashSet<string> _audiences;
+
         /// <summary>
-        /// Gets or sets the value of the 'audience' claim.
+        /// Gets or sets the value of the 'audience' claim. Will be deprecated in favor of <see cref="Audiences"/> in the next
+        /// major version (8.x).
         /// </summary>
         public string Audience { get; set; }
 
+        /// <summary>
+        /// Gets or sets one or more audiences to include in the token's 'Aud' claim. Automatically removes duplicates and empty,
+        /// null, or whitespace-only strings. Does not use a threadsafe collection.
+        /// </summary>
+        public IEnumerable<string> Audiences {
+            get
+            {
+                // If Audiences isn't set, return null since this will be the behavior once Audience is removed.
+                if (_audiences.IsNullOrEmpty())
+                    return null;
+
+                // If both Audience and Audiences are set, return the union of the two.
+                else if (!string.IsNullOrEmpty(Audience))
+                    return _audiences.Union([Audience]);
+
+                // If only Audiences is set, return it
+                else
+                    return _audiences;
+            }
+            set
+            {
+                _audiences = new HashSet<string>(value);
+                _audiences.RemoveWhere(string.IsNullOrWhiteSpace);
+            }
+        }
+
+        /// <summary>
+        /// Adds an audience to the <see cref="Audiences"/> collection. Won't add duplicate, null, empty, or whitespace-only strings.
+        /// </summary>
+        /// <param name="audience">An audience to be added to the Aud claim</param>
+        public void AddAudience(string audience)
```

Comment:
Made the underlying collection a hashset to make it easy to deal with duplicates, and added this AddAudience method in to take advantage of that ability but it does expand our public API. Is this a good/bad idea? Have any better ones?